sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -110,6 +110,7 @@ public class RxDocumentClientImpl implements AsyncDocumentClient, IAuthorization
     DiagnosticsClientContext {
     private static final String tempMachineId = "uuid:" + UUID.randomUUID();
     private static final AtomicInteger activeClientsCnt = new AtomicInteger(0);
+    private static final Map<String, Integer> clientMap = new HashMap<>();
```

Comment:
Why do we need this to be a static final map? Can't we just create a map in the `diagnosticsClientConfig` directly and leverage that.

Reply:
Same reason why the count for clients (activeClientCount) is being tracked outside of the `diagnosticsClientConfig` - we create a new instance every time we run the diagnostics: 
<img width="441" alt="image" src="https://user-images.githubusercontent.com/30335873/186968470-2694e7eb-0873-4dd9-bd3f-28d478d24c62.png">
We would lose track of how many client instances we have per endpoint if we reset it every time by initializing it in the class.

Reply:
Makes sense, thanks!